tools/nimgrep.nim

```diff
@@ -33,13 +33,21 @@ Options:
   --word, -w          the match should have word boundaries (buggy for pegs!)
   --ignoreCase, -i    be case insensitive
   --ignoreStyle, -y   be style insensitive
+  --nWorkers:N, -n:N  speed up search by N additional workers (threads)
   --ext:EX1|EX2|...   only search the files with the given extension(s),
                       empty one ("--ext") means files with missing extension
   --noExt:EX1|...     exclude files having given extension(s), use empty one to
                       skip files with no extension (like some binary files are)
-  --includeFile:PAT   include only files whose names match the given regex PAT
-  --excludeFile:PAT   skip files whose names match the given regex pattern PAT
-  --excludeDir:PAT    skip directories whose names match the given regex PAT
+  --includeFile:PAT   search only files whose names match the given PATttern
```

Comment:
you changed regex PAT to PATttern:
* shouldn't that be PATtern ?
* not clear what kind of patterns are accepted; regex? peg? unix glob?

Reply:
Now both regex and peg patterns are accepted.
I'm not sure what's the right place to describe this, place for --help should be restricted. Manual page then? I'm pondering about adding additional documentation switch like --tutorial or --examples where some additional information would be provided, especially examples of usage.

Reply:
> Now both regex and peg patterns are accepted.

great.
Adding just line in the `--help` would remove a lot of ambiguities as to what PAT represents.

> Manual page then? 

Further details can be found in `nim doc nimgrep` output, as is standard

Reply:
I expanded --help and structured it better. Hopefully it's much clearer now.